src/keyboard_drag_strategy.ts

```diff
@@ -289,4 +283,77 @@ export class KeyboardDragStrategy extends dragging.BlockDragStrategy {
   override shouldHealStack(e: PointerEvent | undefined): boolean {
     return true;
   }
+
+  /**
+   * Get the previous node in the tree, with loopback to the last
+   * stack on the workspace if needed.
+   *
+   * @param start Where to start traversal.
+   * @param cursor The workspace's cursor
+   * @returns The previous node, or null if there were no valid nodes
+   *     on the workspace.
+   */
+  private getPreviousNode(start: ASTNode, cursor: LineCursor) {
```

Comment:
I guess it's not required by either ts or our current linting rules, but I think it's good to add return types here so it's obvious it might return null.